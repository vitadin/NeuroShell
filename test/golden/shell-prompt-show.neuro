%% Shell prompt show command test
\set[_echo_command="true"]
%% Tests shell-prompt-show functionality: displaying current configuration

%% First set up a known prompt configuration
\shell-prompt[lines=2, line1="${@user}@${@pwd}", line2="neuro❯ "]

%% Test showing the configuration
\shell-prompt-show

%% Change to a different configuration
\shell-prompt[lines=1, line1="simple> "]

%% Show the updated configuration
\shell-prompt-show

%% Test with a more complex multi-line configuration
\shell-prompt[lines=4, line1="┌─[${@time}] Session: ${#session_name:-none}", line2="├─[${@user}@${@pwd}]", line3="├─Model: ${#active_model:-none}", line4="└❯ "]

%% Show the complex configuration
\shell-prompt-show