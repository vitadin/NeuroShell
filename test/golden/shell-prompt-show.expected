Setting _echo_command = true
%%> "\\shell-prompt[lines=2, line1=\"${@user}@${@pwd}\", line2=\"neuro❯ \"]"
Prompt lines set to 2
Prompt line 1 set to: testuser@/test/pwd
Prompt line 2 set to: neuro❯ 
%%> "\\shell-prompt-show"
Shell Prompt Configuration:
==========================
Number of lines: 2

Templates:
  Line 1: testuser@/test/pwd
  Line 2: neuro❯ 

Preview:
  testuser@/test/pwd
  neuro❯ 
%%> "\\shell-prompt[lines=1, line1=\"simple> \"]"
Prompt lines set to 1
Prompt line 1 set to: simple> 
%%> "\\shell-prompt-show"
Shell Prompt Configuration:
==========================
Number of lines: 1

Templates:
  Line 1: simple> 

Preview:
  simple> 
%%> "\\shell-prompt[lines=4, line1=\"┌─[${@time}] Session: ${#session_name:-none}\", line2=\"├─[${@user}@${@pwd}]\", line3=\"├─Model: ${#active_model:-none}\", line4=\"└❯ \"]"
Prompt lines set to 4
Prompt line 1 set to: ┌─[12:00:00] Session: 
Prompt line 2 set to: ├─[testuser@/test/pwd]
Prompt line 3 set to: ├─Model: 
Prompt line 4 set to: └❯ 
%%> "\\shell-prompt-show"
Shell Prompt Configuration:
==========================
Number of lines: 4

Templates:
  Line 1: ┌─[12:00:00] Session: 
  Line 2: ├─[testuser@/test/pwd]
  Line 3: ├─Model: 
  Line 4: └❯ 

Preview:
  ┌─[12:00:00] Session: 
  ├─[testuser@/test/pwd]
  ├─Model: 
  └❯ 